code/game/objects/effects/fluids.dm

```diff
@@ -9,82 +9,65 @@
 	layer = FLY_LAYER
 	alpha = 0
 	color = COLOR_LIQUID_WATER
-
-	var/last_slipperiness = 0
-	var/last_flow_strength = 0
-	var/last_flow_dir = 0
 	var/update_lighting = FALSE
 
-/obj/effect/fluid/Initialize()
+/obj/effect/fluid_overlay/Initialize()
 	atom_flags |= ATOM_FLAG_OPEN_CONTAINER
 	icon_state = ""
-	create_reagents(FLUID_MAX_DEPTH)
 	. = ..()
 	var/turf/simulated/T = get_turf(src)
 	if(!isturf(T) || !T.CanFluidPass())
 		return INITIALIZE_HINT_QDEL
 	if(istype(T))
 		T.unwet_floor(FALSE)
 
-/obj/effect/fluid/airlock_crush()
+/obj/effect/fluid_overlay/airlock_crush()
 	qdel(src)
 
-/obj/effect/fluid/Move()
+/obj/effect/fluid_overlay/Move()
 	PRINT_STACK_TRACE("A fluid overlay had Move() called!")
 	return FALSE
 
-/obj/effect/fluid/on_reagent_change()
-	..()
-
-	if(reagents?.total_volume)
-		var/decl/material/primary_reagent = reagents.get_primary_reagent_decl()
-		if(primary_reagent)
-			last_slipperiness = primary_reagent.slipperiness
-
-	ADD_ACTIVE_FLUID(src)
-	for(var/checkdir in global.cardinal)
-		var/obj/effect/fluid/F = locate() in get_step(loc, checkdir)
-		if(F)
-			ADD_ACTIVE_FLUID(F)
-	update_lighting = TRUE
-	update_icon()
-
-/obj/effect/fluid/Destroy()
-	var/turf/simulated/T = get_turf(src)
+/obj/effect/fluid_overlay/Destroy()
```

Comment:
I don't understand why the turf update logic is in the Destroy() override for the visual-only overlay...